php/libraries/TimePoint.class.inc

```diff
@@ -488,7 +488,7 @@ class TimePoint
      */
     function getScreeningStatus(): string
     {
-        return $this->_timePointInfo["Screening"];
+        return $this->_timePointInfo["Screening"] ?? '';
```

Comment:
Why are TimePoint.class.inc and User.class.inc changing?

Reply:
when the tool is run it uses these classes and this suppresses warnings in some instances.

However, I'm not too sure about the user class warnings... I feel like instead of suppressing these we might wanna fix the configuration issue ?

Reply:
by configuration issue I mean that these generally cause warnings when the "user" making changes is not registered in the DB. I would instead register the user in the database in this case

Reply:
I don't see why that would be related to running phpcs on it. If they're required it should be a different PR. The user ones in particular look rather dangerous and likely to introduce bugs down the line.

Reply:
I would assume that they came p while TESTING after running phpcs on them lol.

As for the user ones I agree.

Reply:
The User and Timepoint class were change to prevent fatal errors (not warnings) when the function returns NULL, because the function type does not allow NULL values.

Reply:
They should never be null.

Reply:
I think I found one scenario where a value can be NULL. Is it valid?

```
php fix_timepoint_date_problems.php diagnose 587630
Candidate: 587630 
 Timepoint V1 ; SubProjectID: 1 ; Effective DOB: 2016-05-01 ; (SessionID): 1053 
PHP Fatal error:  Uncaught TypeError: Return value of TimePoint::getScreeningStatus() must be of the type string, null returned in /var/www/html/loris-core/php/libraries/TimePoint.class.inc:491
Stack trace:
#0 /var/www/html/loris-core/tools/fix_timepoint_date_problems.php(661): TimePoint->getScreeningStatus()
#1 /var/www/html/loris-core/tools/fix_timepoint_date_problems.php(286): diagnose()
#2 {main}
  thrown in /var/www/html/loris-core/php/libraries/TimePoint.class.inc on line 491
```

Reply:
Found it, User getUsername() throws an exeption here:
```
if ($user->getUsername() == null) {
        throw new LorisException(
            "Error: Database user named " . getenv('USER')
            . " does not exist. Please create and then retry script\n"
        );
    }
```

I can wrap the statement in a try/catch instead.

Reply:
Just change the check to `if($user === null)`.. the body of the if statement is clearly trying to throw an exception in the case where the user doesn't exist.

Reply:
The exception happens int this part `$user->getUsername()`

Reply:
The user exists, it's an AnonymousUser.

Reply:
Thank you for your suggestion but I don't think it will help. The user is set, it's an AnonymousUser.

Reply:
`(if (is_a($user, 'LORIS\AnonymousUser')) {` will work. I can create another PR for discussing the User fatal errors.

Reply:
Okay, that's a little trickier, maybe our User::singleton needs to be smarter about how it detects an anonymous user.. but `if ($user instanceof AnonymousUser)` still seems safer than this change to the User class.

(I just realized this thread is on `getScreeningStatus`.. the screening status coalesce is valid and much less dangerous than the user class changes.)

Reply:
Thank you @driusan , it's fixed!

Reply:
User class discussion moved to another PR: #7225